base/twiceprecision.jl

```diff
@@ -317,23 +421,31 @@ function *(x::TwicePrecision{<:IEEEFloat}, v::Integer)
 end
 *(v::Number, x::TwicePrecision) = x*v
 
-function *(x::TwicePrecision{T}, y::TwicePrecision{T}) where {T}
-    zh, zl = mul12(x.hi, y.hi)
-    ret = TwicePrecision{T}(canonicalize2(zh, (x.hi * y.lo + x.lo * y.hi) + zl)...)
-    ifelse(iszero(zh) | !isfinite(zh), TwicePrecision{T}(zh, zh), ret)
-end
+*(x::TwicePrecision{T}, y::TwicePrecision{T}) where {T<:AbstractFloat} =
+    let zh = x.hi * y.hi
```

Comment:
This ought to be written as a propert `function`, to match the rest of the style of Base.